sinks/signalfx/signalfx_test.go

```diff
@@ -661,13 +662,13 @@ func TestSignalFxExtractTokensFromResponse(t *testing.T) {
 // mockHandler cycles through a slice of predefined HTTP responses
 type mockHandler struct {
 	responses []string
-	index     int
+	index     int64
 }
 
 func (m *mockHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	resp.WriteHeader(http.StatusOK)
-	resp.Write([]byte(m.responses[m.index%len(m.responses)]))
-	m.index++
+	resp.Write([]byte(m.responses[atomic.LoadInt64(&m.index)%int64(len(m.responses))]))
```

Comment:
nit: I recommend separate this byte array initialization from the `resp.Write` for readability